### PR TITLE
Update clang-format requirement and code to version 13

### LINKED ISSUE #2344: clang-format 9.0.0 not available from PyPi for macOS
Recently, `clang-format >= 9.0.0` was added to `environment.yml`.  This leads to failure on macOS, since PyPi provides clang-format wheels for macOS only from 10.0.1 onwards.

So we either need to take this out of the environment.yml file again or hike the clang-format version.

### PR DESCRIPTION
**Pull request by jougs**
This fixes  #2344.

### CODE REVIEW

**Review by terhorstd — APPROVED** [review]
LGTM :+1:
Thanks!

File: doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst [hunk 1]
```diff
@@ -41,7 +41,7 @@ The code has to compile without warnings (in the default settings of the build
 infrastructure). We restrict ourselves to the C++11 standard for a larger support of
 compilers on various cluster systems and supercomputers.
 
-We use clang-format version 9 in our CI.
+We use clang-format version 13 in our CI.
```
Comment (heplesser) [inline doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst:44]:
```suggestion
We use clang-format version 13 in our CI. If your `clang-format` executable is not version 13, you need to specify an executable with version 13 explicitly with the `--clang-format` option to ensure consistency with the NEST CI.
```

**Review by heplesser — CHANGES_REQUESTED** [review]
@jougs Thank you for the PR! I find some details around the versioning a bit confusing, please see my comments and suggestions.

File: doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst [hunk 2]
```diff
@@ -118,8 +118,8 @@ checks on all changed files as we do during the Travis CI tasks.
                                               the NEST Travis CI build and test environment.
 
        --clang-format=exe               The name of the CLANG-FORMAT executable.
-                                        Default: --clang-format=clang-format-9
-                                        Note: CLANG-FORMAT version 9 is required.
+                                        Default: --clang-format=clang-format-13
```
[the remark this answers is not in the document]
Comment (jougs) [reply doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst]: Fixed in a632cdd14.

File: build_support/check_code_style.sh [hunk 3]
```diff
@@ -113,8 +113,8 @@ Options:
                                            the NEST CI build and test environment.
 
     --clang-format=exe               The name of the CLANG-FORMAT executable.
-                                     Default: --clang-format=clang-format-9
-                                     Note: CLANG-FORMAT version 9 is required.
+                                     Default: --clang-format=clang-format-13
+                                     Note: CLANG-FORMAT version 13 is required.
```
[the remark this answers is not in the document]
Comment (jougs) [reply build_support/check_code_style.sh]: Right. Fixed in a632cdd14.

File: environment.yml [hunk 4]
```diff
@@ -91,7 +91,7 @@ dependencies:
   - pip:
     # For testsuite
     - junitparser >= 2
-    - clang-format == 9.0.0
+    - clang-format == 13.0.0
```
[the remark this answers is not in the document]
Comment (jougs) [reply environment.yml]: Fixed in 186311122.

**Review by heplesser — APPROVED** [review]

File: doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (terhorstd) [reply doc/htmldoc/developer_space/guidelines/coding_guidelines_check.rst]: actually the default is just `clang-format`, as the pip-installed one does not have the slot identifier (see `build_support/check_code_style.sh` line 33)

**Pull request merged**